label_trans_selinux.go

```diff
@@ -20,6 +18,28 @@ const timeout = time.Second * 3
 
 const nullChar = "\000"
 
+/* This is a cursed function and I hate it.
+ * The Go devlopers have decided that machine byteorder is an unimportant detail
+ * and 1) force you to specify order when decoding byte arrays, and
+ * 2) don't give you a simple way of figuring out the native endianness
+ * so here we go. Create a pointer and then look at the byte order to set
+ * the interface for reading off of the socket. */
```

Comment:
Can you start this comment with the function name?

nit: I know it's personal pref, but most multi-line comments I've seen are two slashes on each line:

```
// setNativeEndian ....
// ...
// ...
```